### PR TITLE
adding custom metadata columns

### PR DESCRIPTION
**Pull request by selimonat**
this adds custom metadata columns defined as input dictionary argument to samples, events, messages data frames.

### CODE REVIEW

File: pyedfread/edf.py [hunk 1]
```diff
@@ -51,6 +55,12 @@ def pread(filename,
     events = pd.DataFrame(events)
     messages = pd.DataFrame(messages)
     samples = pd.DataFrame(np.asarray(samples), columns=edfread.sample_columns)
+    
+    for key, value in meta.items():
```
[the remark this answers is not in the document]
Comment (selimonat) [reply pyedfread/edf.py]: the page you referred says this is compatible with 2 and 3 but inefficient in 2.

`# Python 2 and 3: option 1
for (key, value) in heights.items():    # inefficient on Py2 ...`

[the remark this answers is not in the document]
Comment (nwilming) [reply pyedfread/edf.py]: Right, my bad. Should be fine then.

File: pyedfread/edf.py [hunk 2]
```diff
@@ -51,6 +55,12 @@ def pread(filename,
...
+        events.insert(0,key,value)
+        messages.insert(0,key,value)
+        samples.insert(0,key,value)
```
[the remark this answers is not in the document]
Comment (selimonat) [reply pyedfread/edf.py]: This unfortunately doesn't work on python 2. I tested the insert method in both 2 and 3  and the code overall in both 2 and 3, it passes.

**Comment by selimonat** [comment]
see my individual comments. The code at this stage is compatible with both 2 and 3. Is there something else I could do at this stage?

File: pyedfread/edf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nwilming) [reply pyedfread/edf.py]: Samples, events and messages are pandas DataFrames so `samples.loc[:, key] = value` really should work on py2 and py3. I've just double checked this locally and it works for me. 

Can you maybe send the error message that you get with py2?

[the remark this answers is not in the document]
Comment (selimonat) [reply pyedfread/edf.py]:
```/home/onat/anaconda2/envs/py36/lib/python3.6/site-packages/pandas/core/indexing.py in _setitem_with_indexer(self, indexer, value)
    329                         if any([not l for l in len_non_info_axes]):
    330                             if not is_list_like_indexer(value):
--> 331                                 raise ValueError("cannot set a frame with no "
    332                                                  "defined index and a scalar")
    333                             self.obj[key] = value

ValueError: cannot set a frame with no defined index and a scalar
```

[the remark this answers is not in the document]
Comment (nwilming) [reply pyedfread/edf.py]: I still don't understand. If I clone your repository, install it and change the insert to .loc[:, key] = value it works for me without raising an error. In any case, we can leave this as is and I investigate offline.

**Comment by selimonat** [comment]
This works with no problem with .loc[:,key] = value
In [9]: samples, events, messages =
edf.pread("/home/onat/Documents/Code/Python/pyedfread/SUB001.EDF",ignore_samples=True,filter='all',meta={'subj':[12]})

However, when the meta dict doesn't have value of dtype list, it crashes.
In [10]: samples, events, messages =
edf.pread("/home/onat/Documents/Code/Python/pyedfread/SUB001.EDF",ignore_samples=True,filter='all',meta={'subj':"12"})
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-10-f0acee2c3ba1> in <module>()
----> 1 samples, events, messages =
edf.pread("/home/onat/Documents/Code/Python/pyedfread/SUB001.EDF",ignore_samples=True,filter='all',meta={'subj':"12"})

/home/onat/anaconda2/lib/python2.7/site-packages/pyedfread/edf.pyc in
pread(filename, ignore_samples, filter, split_char, trial_marker, meta)
     60         events.loc[:,key] = value
     61         messages.loc[:,key]=value
---> 62         samples.loc[:,key] = value
     63         #events.insert(0,key,value)
     64         #messages.insert(0,key,value)

/home/onat/anaconda2/lib/python2.7/site-packages/pandas/core/indexing.pyc
in __setitem__(self, key, value)
    139             key = com._apply_if_callable(key, self.obj)
    140         indexer = self._get_setitem_indexer(key)
--> 141         self._setitem_with_indexer(indexer, value)
    142
    143     def _has_valid_type(self, k, axis):

/home/onat/anaconda2/lib/python2.7/site-packages/pandas/core/indexing.pyc
in _setitem_with_indexer(self, indexer, value)
    289                         if any([not l for l in len_non_info_axes]):
    290                             if not is_list_like_indexer(value):
--> 291                                 raise ValueError("cannot set a
frame with no "
    292                                                  "defined index and
a scalar")
    293                             self.obj[key] = value

ValueError: cannot set a frame with no defined index and a scalar

In [11]:

>

**Pull request merged**